elementary/scratch
# No restore failed doc

Fixes #81.
Fixes #5 

This branch delays various operations until a document has successfully loaded asynchronously rather than assuming success. This includes adding the document to the list to be saved and restored, thereby fixing the linked issue.

Files are checked to be of the supertype "text/plain" before opening to avoid problems opening inappropriate files.

Speed of loading of large files is increased by first loading into a separate buffer to avoid continuous reformatting of the view.

The load error dialog is minimally fixed in passing (but needs redesign).

TO TEST:

Attempt to load a binary or other non-text file under various conditions.  The app should not hang nor try to restore the file.
